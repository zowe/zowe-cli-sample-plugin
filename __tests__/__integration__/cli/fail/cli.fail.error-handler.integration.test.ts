/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import {runCliScript} from "../../../__src__/TestUtils";
import {ITestEnvironment} from "../../../__src__/environment/doc/response/ITestEnvironment";
import {TestEnvironment} from "../../../__src__/environment/TestEnvironment";

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment;

describe("zowe-cli-sample fail error-handler command", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            installPlugin: true,
            testName: "fail_command",
            skipProperties: true
        });
    });
    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });
    it("should fail the handler", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/fail_error_handler.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
        expect(response.stdout.toString()).toMatchSnapshot();
    });
});
