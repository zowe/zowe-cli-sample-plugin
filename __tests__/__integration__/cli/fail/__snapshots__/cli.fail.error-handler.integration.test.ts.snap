// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zowe-cli-sample fail error-handler command should fail the handler 1`] = `""`;

exports[`zowe-cli-sample fail error-handler command should fail the handler 2`] = `
"{
  \\"success\\": false,
  \\"exitCode\\": 1,
  \\"message\\": \\"\\\\\\"zowe zowe-cli-sample zcsp fail error-handler\\\\\\" command failed!\\",
  \\"stdout\\": \\"\\",
  \\"stderr\\": \\"Command Error:\\\\n\\\\\\"zowe zowe-cli-sample zcsp fail error-handler\\\\\\" command failed!\\\\n\\",
  \\"data\\": {},
  \\"error\\": {
    \\"msg\\": \\"\\\\\\"zowe zowe-cli-sample zcsp fail error-handler\\\\\\" command failed!\\"
  }
}"
`;
